src/WebJobs.Script/Host/ScriptHost.cs

```diff
@@ -497,63 +507,21 @@ internal async Task InitializeFunctionDescriptorsAsync(IEnumerable<FunctionMetad
                 functions = await GetFunctionDescriptorsAsync(functionMetadata, _descriptorProviders);
                 _logger.LogDebug("Function descriptors created.");
             }
-
             Functions = functions;
         }
 
-        private async Task InitializeWorkersAsync()
+        private void InitializeLanguageWorkerFunctionDispatcher()
         {
-            var serverImpl = new FunctionRpcService(EventManager, _logger);
-            await InitializeRpcServiceAsync(new GrpcServer(serverImpl, ScriptOptions.MaxMessageLengthBytes));
-
-            var processFactory = new DefaultWorkerProcessFactory();
-            try
-            {
-                _processRegistry = ProcessRegistryFactory.Create();
-            }
-            catch (Exception e)
+            _languageWorkerChannelManager.InitializeMetricsLogger(_metricsLogger);
+            ILanguageWorkerChannel initializedChannel = _languageWorkerChannelManager.GetWebHostChannel(_currentRuntimelanguage);
+            _functionDispatcher.CurrentLanguageRuntime = _currentRuntimelanguage;
```

Comment:
We should try to avoid setting this property on the live service, as this introduces a potential race/ordering issue (any code inspecting this property will not have the correct information).

We should have a single source of truth for this information and it shouldn't change during service initialization within this scope. So we can remove the `_currentRuntimelanguage` from the ScriptHost and have the dispatcher set its property (and change that to read only) at initialization.

Reply:
done. Removed dependency on LanguageWorkerChannelManager in scripthost